### PR TITLE
stats: Recent lookups implementation

### PR DESCRIPTION
**Pull request by jmarantz**
Description: Split from #8116 -- add and test an class to hold recent stats lookups by name. It makes a tiny LRU cache holding only a count of recent hits for a particular name. Once this and #8369 are merged then #8116 can pull these together to show recent-lookups as an admin endpoint.
Risk Level: low -- this is a new class that isn't used yet
Testing: //test/... and speed test
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by jmarantz** [comment]
@envoyproxy/maintainers can someone self-assign for a first pass? It's really a very simple specialized LRU cache for the purpose of tracking down recently/frequently stat-names looked up by name, so they can be memoized at startup or otherwise dealt with. This is just a new class and its tests so it should be easy to review.

File: source/common/stats/recent_lookups.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   * @return The configured capacity.
+   */
+  uint64_t capacity() const { return capacity_; }
+
+private:
+  void evictOne();
+
+  struct ItemCount {
+    std::string item_;
+    int64_t count_;
```
Comment (mattklein123) [inline source/common/stats/recent_lookups.h]: nit: uint64_t

File: source/common/stats/recent_lookups.cc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+void RecentLookups::setCapacity(uint64_t capacity) {
+  capacity_ = capacity;
+  while (capacity_ < list_.size()) {
+    evictOne();
+  }
+}
+
+void RecentLookups::evictOne() {
+  const ItemCount& item_count = list_.back();
```
Comment (mattklein123) [inline source/common/stats/recent_lookups.cc]: nit: assert list and map are non-empty before doing this work?

**Pull request merged**